company_website/tests/test_send_email_utils.py

```diff
@@ -0,0 +1,95 @@
+from django.conf import settings
+from django.core import mail
+from django.test import TestCase
+from parameterized import parameterized
+
+from company_website.constants import ESTIMATE_PROJECT_EMAIL_SUBJECT
+from company_website.constants import ESTIMATE_PROJECT_EMAIL_TEMPLATE_NAME
+from company_website.models import EstimateProjectEmailRecipients
+from company_website.send_emails_utils import _convert_html_to_plain_text
+from company_website.send_emails_utils import _create_messages_to_mail
+from company_website.send_emails_utils import _get_estimate_project_email_recipients
+from company_website.send_emails_utils import send_mail_to_management
+
+
+class EstimateProjectEmailRecipientsTestCase(TestCase):
+    def test_that_when_use_default_recipients_settings_is_enabled_method_returns_default_recipients_list(self):
+        settings.USE_DEFAULT_RECIPIENTS = True
+
+        returned_recipients_list = _get_estimate_project_email_recipients()
+
+        self.assertEqual(returned_recipients_list, settings.DEFAULT_EMAIL_RECIPIENTS)
+
+    def test_that_when_recipient_list_is_empty_method_returns_default_email_recipients_list(self):
+        returned_recipients_list = _get_estimate_project_email_recipients()
+
+        self.assertEqual(returned_recipients_list, settings.DEFAULT_EMAIL_RECIPIENTS)
+
+    def test_that_when_recipient_list_is_not_empty_and_default_recipient_settings_is_disabled_it_return_recipients_list(
+        self,
+    ):
+        defined_recipient = "some_recipient@codepoets.it"
+
+        EstimateProjectEmailRecipients(name=defined_recipient).save()
+
+        returned_recipients_list = list(_get_estimate_project_email_recipients())
+
+        self.assertEqual(returned_recipients_list, [(defined_recipient,)])
+
+
+class SendEmailTestCase(TestCase):
+    def setUp(self) -> None:
+        self.template = ESTIMATE_PROJECT_EMAIL_TEMPLATE_NAME
+        self.template_context = {
+            "name": "John Doe",
+            "email": "email@address.com",
+            "idea_description": "Some description",
+            "privacy_policy_accepted": True,
+        }
+
+        self.expected_mail_body = (
+            "Project Liberation - New estimate project form has been filled in.\n\n"
+            "Form data:\n\n"
+            "Name: John Doe\n\n"
+            "Email address: email@address.com\n\n"
+            "Country:\n\n"
+            "Idea description: Some description\n\n"
+            "Is NDA required:\n\n"
+            "You want to\n\n"
+            "What is your monthly budget?\n\n"
+            "How much time do you have?\n\n"
+            "Do you want us to design your product?\n\n"
+            "How did you find us?:"
+        )
```

Comment:
This is not expected mail body. It shouldn't show fields which are not fulfilled in the form.

Reply:
Done. I fixed that by replace `""` with `null` in if statements , in **estimate_project_email.html** file

Reply:
Done. I forgot to add fields with empty string values to template context data.